homeassistant/components/alarmdecoder/alarm_control_panel.py

```diff
@@ -163,6 +165,7 @@ def device_state_attributes(self):
         """Return the state attributes."""
         return {
             "ac_power": self._ac_power,
+            "alarm_event_occurred": self._alarm_event_occurred,
```

Comment:
Reading the docs PR, it sounds like this attribute is correlated to the triggered state of the alarm panel. Is that true, and if so why do we need this attribute?

Reply:
It’s the “alarm_sounding” parameter that’s associated with the current triggered state of the alarm panel. “alarm_event_occurred” is the parameter that’s associated with the “ALARM XX” message on the actual panel that’s supposed to tell you that the alarm was triggered on such and such zone, and needs to be acknowledged by the user before arming is possible.

Reply:
Ok. But if the alarm state was triggered, how does it change state from triggered without the user disarming the alarm?

Reply:
here's a quick 2-minute video that might help. if an alarm is triggered on a Honeywell system, the user must disarm the alarm system _twice_ before being able to arm again. the first time you disarm the system, it will disable the sirens/lights that are typically activated when the system is triggered. BUT, the alarm keypad display will still say something like "ALARM 01 FRONT DOOR" which shows what zone triggered the alarm initially. as long as this message is on the keypad, the alarm system can't be re-armed. in order to be able to re-arm the system again, that message has to be cleared by disarming the system once more.

https://www.youtube.com/watch?v=U4eY8x0Nelk

Reply:
What state will the alarm be in after the first disarm before the second disarm?

Reply:
i just triggered my alarm and tested it. the alarm is in the `disarmed` state after the first disarm, but before the second disarm.

Reply:
Ok, then it sounds like this new attribute adds information that is needed. Thanks for the clarification and testing. :+1:

Reply:
Normally we don't add the item at all to the state attributes if the value of the item is `None`. But since we don't have that check yet for any of the attributes here, I think we can wait with that change since it would be a breaking change.